src/ace/managers/mouse.c

```diff
@@ -1,126 +1,128 @@
-	logBlockBegin("mouseSetPointer(pCursor: %p, wHeight: %d, wWidth, %d, wOffsetX: %d, wOffsetY: %d)", pCursor, wHeight, wWidth, wOffsetX, wOffsetY);
-#ifdef AMIGA
-	SetPointer(g_sWindowManager.pWindow, pCursor, wHeight, wWidth, wOffsetX, wOffsetY);
-#endif // AMIGA
-	logBlockEnd("mouseSetPointer");
-}
+	if(ubPortFlags & MOUSE_PORT_2) {
+		uwPotMask |= BV(15) | BV(14) | BV(13) | BV(12);
+		memset(&g_sMouseManager.pMouses[MOUSE_PORT_2], 0, sizeof(tMouse));
+		g_sMouseManager.pMouses[MOUSE_PORT_2].uwX = (g_sMouseManager.uwHiX - g_sMouseManager.uwLoX) >> 1;
+		g_sMouseManager.pMouses[MOUSE_PORT_2].uwY = (g_sMouseManager.uwHiY - g_sMouseManager.uwLoY) >> 1;
+	}
+	custom.potgo = (custom.potinp & (0xFFFF ^ uwPotMask)) | uwPotMask;
 
-void mouseResetPointer(void) {
-#ifdef AMIGA
-	ClearPointer(g_sWindowManager.pWindow);
+	// Amiga Hardware Reference Manual suggests that pos should be polled every
+	// vblank, so there could be some interrupt init.
 #endif // AMIGA
 }
 
+void mouseDestroy(void) {
 #ifdef AMIGA
-void _mouseDo(struct InputEvent *pEvent) {
-	g_sMouseManager.pInputIO->io_Data = (APTR) pEvent;
-	g_sMouseManager.pInputIO->io_Length = sizeof(struct InputEvent);
-	g_sMouseManager.pInputIO->io_Command = IND_WRITEEVENT;
-
-	DoIO((struct IORequest *) g_sMouseManager.pInputIO);
-}
+	// Should mouse manager be interrupt driven, interrupt handler deletion will
+	// be here.
+	custom.potgo = g_sMouseManager.uwPrevPotGo;
 #endif // AMIGA
-
-void mouseSetPosition(UWORD uwNewX, UWORD uwNewY) {
-	mouseMoveBy(uwNewX - mouseGetX(), uwNewY - mouseGetY());
 }
 
-void mouseMoveBy(WORD wDx, WORD wDy) {
-#ifdef AMIGA
-	struct InputEvent __chip sEvent;
-
-	sEvent.ie_Class = IECLASS_POINTERPOS;
-	sEvent.ie_Code = IECODE_NOBUTTON;
-	sEvent.ie_Qualifier = IEQUALIFIER_RELATIVEMOUSE;
-	sEvent.ie_X = wDx;
-	sEvent.ie_Y = wDy;
-
-	_mouseDo(&sEvent);
-#endif // AMIGA
+void mouseSetBounds(UWORD uwLoX, UWORD uwLoY, UWORD uwHiX, UWORD uwHiY) {
+	g_sMouseManager.uwLoX = uwLoX;
+	g_sMouseManager.uwLoY = uwLoY;
+	g_sMouseManager.uwHiX = uwHiX;
+	g_sMouseManager.uwHiY = uwHiY;
 }
 
-void mouseClick(UBYTE ubMouseCode) {
+void mouseProcess(void) {
+	// Even if whole Amiga process will be moved to vbl interrupt, other platforms
+	// will prob'ly use this fn anyway
 #ifdef AMIGA
-	struct InputEvent *pEvent = memAllocChipFlags(sizeof(struct InputEvent), MEMF_PUBLIC | MEMF_CLEAR);
-
-	pEvent->ie_Class = IECLASS_RAWMOUSE;
-	pEvent->ie_Code = ubMouseCode;
-
-	_mouseDo(pEvent);
-
-	memFree(pEvent, sizeof(struct InputEvent));
-#endif // AMIGA
-}
+	volatile UWORD * const pCiaAPra = (UWORD*)((UBYTE*)0xBFE001);
+
+	// Deltas are signed bytes even though underflows and overflows may occur.
+	// It is expected behavior since it is encouraged in Amiga HRM as means to
+	// determine mouse movement direction which takes into account joyxdat
+	// underflows and overflows.
+
+	if(g_sMouseManager.ubPortFlags & MOUSE_PORT_1) {
```

Comment:
This IFs should be totally a function with few variables. Only 4 hardcoded elements varies.

Reply:
Done in dab7e1d